heliocron: init at 0.8.1

TheColorman opened the pull request:
## Description of changes

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->
[heliocron](https://github.com/mfreeborn/heliocron) is a program that delays execution of subsequent commands  based on the sunset/sunrise at a given location.

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

TheColorman committed:
maintainers: add TheColorman

TheColorman commented:
I'd love to try and build it for aarch64-linux, but I'm not sure how to do so (and test the result) on an x86_64-linux machine. According to the heliocron README, aarch64 support should be good.

Sigmanificient commented on pkgs/by-name/he/heliocron/package.nix:
```diff
@@ -0,0 +1,33 @@
+
+  src = fetchFromGitHub {
+    owner = "mfreeborn";
+    repo = pname;
```
```suggestion
    repo = "heliocron";
```

TheColorman replied:
Is there documentation stating that a hardcoded string is preffered to pname? I tried to read through all of the contributing documentation and didn't see any, so I'd love to know if there's something else I've missed!

kirillrdy replied:
pname is a bit special, here is a tracking issue https://github.com/NixOS/nixpkgs/issues/277994

kirillrdy commented on pkgs/by-name/he/heliocron/package.nix:
```diff
@@ -0,0 +1,33 @@
+
+  cargoHash = "sha256-RjYE2TBvHnBgiwJi/Ys+wAmJoMHJuimcD07FOqwkcis=";
+
+  meta = with lib; {
```
`with lib;` is also considered harmful https://github.com/NixOS/nixpkgs/issues/208242

TheColorman replied:
Makes perfect sense, I mostly copied the `package.nix` contents from a different package (that I later found it was apparently in the "deprecated" `pkgs/applications/` directory) so I assumed it was standard.

kirillrdy commented:
> I'd love to try and build it for aarch64-linux, but I'm not sure how to do so (and test the result) on an x86_64-linux machine. According to the heliocron README, aarch64 support should be good.

i use binfmt on x86 to build and run aarch64-linux

https://search.nixos.org/options?channel=24.05&from=0&size=50&sort=relevance&type=packages&query=boot.binfmt.emulatedSystems


from CI you can see that heliocron builds fine on aarch64-linux 
![image](https://github.com/user-attachments/assets/fb57ab03-b37a-411f-9dcd-1cc669e6b5f4)

kirillrdy commented:
```
[nix-shell:~/.cache/nixpkgs-review/pr-338936]$ file ./results/heliocron/bin/heliocron 
./results/heliocron/bin/heliocron: ELF 64-bit LSB pie executable, ARM aarch64, version 1 (SYSV), dynamically linked, interpreter /nix/store/5qfsgr600bry4a6431w311jv4a73v8js-glibc-2.39-52/lib/ld-linux-aarch64.so.1, for GNU/Linux 3.10.0, not stripped
```

```
[nix-shell:~/.cache/nixpkgs-review/pr-338936]$ ./results/heliocron/bin/heliocron 
heliocron 0.8.1
A simple program for calculating sunrise, sunset and related times, which can be integrated with
cron to trigger other programs to run when these events occur

USAGE:
    heliocron [OPTIONS] <SUBCOMMAND>

OPTIONS:
    -d, --date <DATE>              Set the date for which the calculations should be run. If
                                   specified, it should be in 'yyyy-mm-dd' format, otherwise it
                                   defaults to the the current local date [default: 2024-09-02]
    -h, --help                     Print help information
    -l, --latitude <LATITUDE>      Set the latitude in decimal degrees. Positive values to the
                                   north; negative values to the south. Defaults to '51.4769' if not
                                   otherwise specified here or in ~/.config/heliocron.toml
    -o, --longitude <LONGITUDE>    Set the longitude in decimal degrees. Positive values to the
                                   east; negative values to the west. Defaults to '-0.0005' if not
                                   otherwise specified here or in ~/.config/heliocron.toml
    -t, --time-zone <TIME_ZONE>    Set the time zone. If specified, it should be in the format
                                   '[+/-]HH:MM', otherwise it defaults to the current local time
                                   zone [default: +10:00]
    -V, --version                  Print version information

SUBCOMMANDS:
    help      Print this message or the help of the given subcommand(s)
    poll      Display real time data pertaining to the Sun at the current local time
    report    Produce a full set of sunrise, sunset and other related times for the given date
                  and location
    wait      Set a delay timer which will expire when the chosen event (+/- optional offset)
                  occurs
```

TheColorman commented:
> i use binfmt on x86 to build and run aarch64-linux

Oh nice, that's really cool! Thanks.

kirillrdy commented on pkgs/by-name/he/heliocron/package.nix:
```diff
@@ -0,0 +1,33 @@
+  cargoHash = "sha256-RjYE2TBvHnBgiwJi/Ys+wAmJoMHJuimcD07FOqwkcis=";
+
+  meta = {
+    description = "Command line application written in Rust capable of delaying execution of other programs for time periods relative to sunrise and sunset";
```
is it possible to have shorter description, currently `longDescription` is same length as `description`

we can borrom from arch, `Execute tasks relative to sunset, sunrise and other solar events` ?

TheColorman replied:
Sure, I just copied the description from the Github repo, I'll shorten it.

kirillrdy commented:
@TheColorman can you please squash your commits into `heliocron: init at 0.8.1`, so there should be just two commits

 - maintainers: add TheColorman
 - heliocron: init at 0.8.1

usually multiple commits is fine, but for init PR's and when commits are so small, it's better to reduce noise.

TheColorman committed:
heliocron: init at 0.8.1

Command line application for delaying execution of other commands relative to sunrise and sunset

https://github.com/mfreeborn/heliocron

kirillrdy commented:
thank you !

TheColorman commented:
Guess I can call myself a nixpkgs maintainer now 😅
